frontend/src/pages/RecipientRecord/pages/MergeGoals/index.js

```diff
@@ -281,6 +285,20 @@ export default function MergeGoals({
     </button>
   );
 
+  const onSubmit = async (data) => {
+    try {
+      setIsAppLoading(true);
+      const mergedGoals = await mergeGoals(data.selectedGoalIds, data.finalGoalId);
+      const goalIds = mergedGoals.map((g) => g.id);
+      setIsAppLoading(false);
+      history.push(`${backPath}`, { mergedGoals: goalIds });
+    } catch (e) {
+      setValidation('Unable to merge goals');
+      setIsAppLoading(false);
```

Comment:
Should this be in finally or will redirect happen if its all good.

Reply:
Tried it in the finally when I was writing this, didn't work exactly right